src/osd/modules/render/drawbgfx.cpp

```diff
@@ -382,6 +386,36 @@ bool video_bgfx::init_bgfx_library(osd_window &window)
 }
 
 
+//============================================================
+//  Helper for creating a wayland window
+//============================================================
+
+#if defined(SDLMAME_USE_WAYLAND)
+wl_egl_window *create_wl_egl_window(SDL_Window *window, struct wl_surface *surface)
+{
+	if (!surface)
+		{
+			osd_printf_error("Wayland surface missing, aborting\n");
+			return nullptr;
+		}
+	wl_egl_window *win_impl = (wl_egl_window *)SDL_GetWindowData(window, "wl_egl_window");
+	if (!win_impl)
+	{
+		int width, height;
+		SDL_GetWindowSize(window, &width, &height);
+		win_impl = wl_egl_window_create(surface, width, height);
+		if (!win_impl)
+			{
+				osd_printf_error("Creating wayland window failed\n");
+				return nullptr;
+			}
+		SDL_SetWindowData(window, "wl_egl_window", win_impl);
+	}
+	return win_impl;
+}
```

Comment:
Indentation is off in two places – you’ve bumped two levels for some of the `if` statements.  The braces should be at the level of the `if` statement itself.

Reply:
Done